tests/smoke_tests/checkers.py

```diff
@@ -0,0 +1,206 @@
+        raise NotImplementedError
+
+    @abstractmethod
+    def check(self, log_line: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def found_all_metrics(self) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    def to_dict(self) -> Dict[str, Union[str, float]]:
+        raise NotImplementedError
+
+
+class LossChecker(MetricChecker):
+    def __init__(
+        self,
+        loss: float,
+        loss_type: LossType = LossType.LOSS,
+        metric_type: Optional[MetricType] = None,
+        tolerance: float = DEFAULT_TOLERANCE,
+    ):
+        self.loss = loss
+        self.loss_type = loss_type
+        self.metric_type = metric_type
+        self.tolerance = tolerance
+        self.found_loss = False
+
+    def found_all_metrics(self) -> bool:
+        return self.found_loss
+
+    def should_check(self, log_line: str) -> bool:
+        if self.loss_type == LossType.LOSS:
+            return "losses_distributed" in log_line
+
+        # all other loss types:
+        assert self.metric_type is not None
+        return f"Client {self.metric_type.value} Losses" in log_line
+
+    def check(self, log_line: str) -> None:
+        losses = self.parse_losses(log_line)
+        has_loss = any([approx(loss, abs=self.tolerance) == self.loss for loss in losses])
+
+        if has_loss:
+            self.found_loss = True
+
+    def parse_losses(self, log_line: str) -> List[float]:
+        try:
+            if self.loss_type == LossType.LOSS:
+                keyword = "losses_distributed"
+                if keyword in log_line:
+                    losses_string = log_line.split(keyword)[-1]
+                    losses_list = eval(losses_string)
+                    return [loss_tuple[1] for loss_tuple in losses_list]
+            else:
+                keyword = f"{self.loss_type.value}:"
+                if keyword in log_line:
+                    after_keyword = log_line.split(keyword)[-1]
+                    loss_string = after_keyword[: after_keyword.find("\t")]
+                    return [eval(loss_string)]
+        except Exception:
+            # if any parsing error occurs, return empty
+            return []
+        return []
+
+    def to_dict(self) -> Dict[str, Union[str, float]]:
+        return {
+            "loss": self.loss,
+            "loss_type": self.loss_type.value,
+            "metric_type": self.metric_type.value if self.metric_type is not None else "",
+            "tolerance": self.tolerance,
+        }
+
+
+class AccuracyChecker(MetricChecker):
+    def __init__(
+        self,
+        accuracy: float,
+        metric_type: MetricType,
+        scope: MetricScope = MetricScope.PREDICTION,
+        tolerance: float = DEFAULT_TOLERANCE,
+    ):
+        self.metric_type = metric_type
+        self.accuracy = accuracy
+        self.scope = scope
+        self.tolerance = tolerance
+        self.found_accuracy = False
```

Comment:
Super minor: You could make this property something more generic like `found_target` and put it in the `MetricChecker` base class and move it out of the child implementations.